mewarpx/mewarpx/utils_store/util.py

```diff
@@ -370,3 +370,82 @@ def plasma_Debye_length(T, n):
     return np.sqrt(
         constants.epsilon_0 * constants.kb_J * T / (n * constants.e**2)
     )
+
+
+def get_vel_vector(v_mag):
+    """Function that returns a random velocity vector given a magnitude.
+    The random point on a unit sphere is found according to
+    https://math.stackexchange.com/questions/44689/how-to-find-a-random-
+    axis-or-unit-vector-in-3d.
+    """
+    theta = np.random.rand(*v_mag.shape) * 2.0 * np.pi
+    z = 2.0 * np.random.rand(*v_mag.shape) - 1.0
+    vel_vectors = np.zeros(v_mag.shape + (3,))
+    vel_vectors[..., 0] = np.sqrt(1.0 - z**2) * np.cos(theta)
+    vel_vectors[..., 1] = np.sqrt(1.0 - z**2) * np.sin(theta)
+    vel_vectors[..., 2] = z
+    vel_vectors = v_mag[..., None] * vel_vectors
+
+    # assert(np.allclose(v_mag, np.sqrt(np.sum(vel_vectors**2, axis=1))))
+
+    return vel_vectors
+
+
+def interpolate_from_grid(coords, grid):
```

Comment:
I expect this function can be optimized some, perhaps by using Numba. If we can get it sufficiently fast to be competitive with the C++ implementation of the same function (in `/Source/Particles/Gather/ScalarFieldGather.cpp`) we can move our Schottky enhancement implementation completely back to Python which would save us needing to inject an intermediate set of particles just to have them distributed to the proper tiles before upweighting them.